### PR TITLE
fix parallel_input flag for Linear1D_Col gather_output

### PR DESCRIPTION
**Pull request by Wesley-Jzy**

### CODE REVIEW

File: colossalai/nn/layer/parallel_1d/layers.py [hunk 1]
```diff
@@ -302,7 +302,10 @@ def __init__(self,
         with seed(ParallelMode.TENSOR):
             self.reset_parameters(weight_initializer, bias_initializer)
         self._set_tensor_parallel_attributes()
-        set_parallel_input(True)
+        if self.gather_output:
```
Comment (kurisusnowdeng) [inline colossalai/nn/layer/parallel_1d/layers.py:305]: How about `set_parallel_input(not self.gather_output)`

**Review by kurisusnowdeng — CHANGES_REQUESTED** [review]

File: colossalai/nn/layer/parallel_1d/layers.py (hunk 1 above)
Comment (Wesley-Jzy) [reply colossalai/nn/layer/parallel_1d/layers.py:305]: LGTM.
Also, what about
```
is_parallel_output = not self.gather_output
set_parallel_input(is_parallel_output)
```? Make it easier to understand and scale in the future as well as to keep simplicity.

Comment (Wesley-Jzy) [reply colossalai/nn/layer/parallel_1d/layers.py]: @kurisusnowdeng Done.

**Comment by feifeibear** [comment]
#600

**Pull request merged**